server/game/gameSystems/ResourceCardSystem.ts

```diff
@@ -0,0 +1,76 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, EffectName, Location, RelativePlayer, WildcardCardType } from '../core/Constants';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
+import { ready } from './GameSystemLibrary';
+import * as Contract from '../core/utils/Contract';
+
+export interface IResourceCardProperties extends ICardTargetSystemProperties {
+    // TODO: remove completely if faceup logic is not needed
+    // faceup?: boolean;
+    targetPlayer?: RelativePlayer; // TODO: this might be needed for Arquitens Assault Cruiser
+    readyResource?: boolean;
+}
+
+export class ResourceCardSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, IResourceCardProperties> {
+    public override readonly name = 'resource';
+    public override targetTypeFilter = [WildcardCardType.Unit, WildcardCardType.Upgrade, CardType.Event];
+
+    protected override defaultProperties: IResourceCardProperties = {
+        // TODO: remove completely if faceup logic is not needed
+        // faceup: false,
+        targetPlayer: RelativePlayer.Self,
+        readyResource: false
+    };
+
+    public eventHandler(event: any, additionalProperties = {}): void {
+        const context = event.context;
+        // TODO: remove this completely if determinmed we don't need card snapshots
+        // event.cardStateWhenMoved = card.createSnapshot();
+        const properties = this.generatePropertiesFromContext(context, additionalProperties) as IResourceCardProperties;
+        // TODO: If we ever need to resource multiple cards at once, this will need an update
+        const card = Array.isArray(properties.target) ? properties.target[0] as Card : properties.target as Card;
+
+        const player = properties.targetPlayer === RelativePlayer.Opponent ? card.controller.opponent : card.controller;
+        player.moveCard(card, Location.Resource);
+        if (properties.readyResource) {
+            context.game.openEventWindow(ready({ target: card }).generateEvent(context.source, context));
```

Comment:
This line it shouldn't be here - what this line currently does is interrupt the flow of resourcing the card to ready the card _in its pre-move location_, since the move hasn't resolved yet. The contingent event code below should be correct

Reply:
I noticed that this doesn't seem to actually be working